### ISSUE TITLE
`AccessorMethodName` check should allow for non-accessor methods

### ISSUE THREAD

**Issue by mockdeep** [bbatsov/rubocop#2509]
We've got methods where we want to assign some data somewhere but aren't setters. For example, we'd like to have a `set_event_context` method that stores some information in the session about the current request. It's not a setter in that we don't pass in something to be assigned, so ideally it would let us get away with the naming in those situations. We end up having to come up with other names for the action, like `assign_event_context` or the like.

**Comment by alexdowad** [comment]
This is already how `AccessorMethodName` works. Look at the source:

```ruby
        def bad_writer_name?(method_name, args)
          method_name.start_with?('set_') && args.to_a.one?
        end
```

As you can see, it doesn't flag method definitions which have no arguments, or more than one argument. For those which have exactly one argument, the preferred style is `self.event_context = obj`.

I think this can be closed.

**Comment by mockdeep** [comment]
Hmm, I guess it's changed more recently than when I ran into it before. The one we were running into issues with last week was something like:

```ruby
def set_error_context(context = {})
  @error_context.merge!(context)
end
```

But I guess we can work around that other ways.

**Comment by mockdeep** [comment]
We were trying to use the same method in a `before_action` and directly. Probably best to split out into 2 methods.
